### PR TITLE
Use real_t in 3D nodes

### PR DESCRIPTION
**Pull request by aaronfranke**
This is similar to other recent `real_t` PRs I've made, but this PR specifically focuses on 3D nodes.

Some notes:

* I made `Camera3D` use `real_t` because `CameraMatrix` already does.

* All code related to time was intentionally left unchanged. Code that works with time should be `double`, not `real_t`, but that would change behavior, so I think it's worth doing in a separate PR.

### CODE REVIEW

File: scene/3d/cpu_particles_3d.cpp [hunk 1]
```diff
@@ -484,7 +484,7 @@ static float rand_from_seed(uint32_t &seed) {
 		s += 2147483647;
 	}
 	seed = uint32_t(s);
-	return float(seed % uint32_t(65536)) / 65535.0;
+	return (seed % uint32_t(65536)) / 65535.0;
```
Comment (KoBeWi) [inline scene/3d/cpu_particles_3d.cpp:487]: Why casting was removed here, but it was changed to `real_t` few lines below? (613)

Comment (aaronfranke) [reply scene/3d/cpu_particles_3d.cpp:487]: `65535.0` is a double literal, so before, the `seed % uint32_t(65536)` was casted from `uint32_t` to `float` to `double`. With the `float` cast removed, the `seed % uint32_t(65536)` is casted from `uint32_t` to `double` directly.

**Review by KoBeWi — APPROVED** [review]
This is mostly the same as #45571, so should be fine I guess.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**